### PR TITLE
multipass: introduce MultipassProvider

### PR DESCRIPTION
**Pull request by cjp256**
Compatibility and cleaning checks will be introduced later.

Depends on #20

### CODE REVIEW

File: craft_providers/multipass/multipass_provider.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    def _configure_instance(
+        self,
+        *,
+        instance: MultipassInstance,
+        base_configuration: Base,
+    ) -> None:
+        """Configure instance.
+
+        Start to instance first to ensure it is started, as well as to cancel
+        any outstanding delay-shtudown request.
```
Comment (sergiusens) [inline craft_providers/multipass/multipass_provider.py:70]:
```suggestion
        Start the instance first to ensure it is started, as well as to cancel
        any outstanding delay-shtudown request.
```

File: craft_providers/multipass/multipass_provider.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                image=image_name,
+            )
+
+        self._configure_instance(
+            instance=instance,
+            base_configuration=base_configuration,
+        )
+        return instance
+
+    def install(self) -> str:
```
Comment (sergiusens) [inline craft_providers/multipass/multipass_provider.py:128]: souldn't this be a static class method?

File: craft_providers/multipass/multipass_provider.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            _install_windows()
+        else:
+            raise errors.MultipassInstallationError(
+                f"unsupported platform {sys.platform!r}"
+            )
+
+        multipass_version, _ = Multipass().wait_until_ready()
+        return multipass_version
+
+    def is_installed(self) -> bool:
```
Comment (sergiusens) [inline craft_providers/multipass/multipass_provider.py:149]: souldn't this be a static class method?

File: craft_providers/multipass/multipass_provider.py (hunk 2 above)
Comment (cjp256) [reply craft_providers/multipass/multipass_provider.py:128]: Probably should just drop the class altogether.

**Pull request closed without merging**